src/effects/VST3/VST3Effect.cpp

```diff
@@ -242,170 +237,6 @@ bool VST3Effect::LoadFactoryPreset(int id, EffectSettings& settings) const
    return true;
 }
 
-namespace
-{
-   unsigned CountChannels(Steinberg::Vst::IComponent* component,
-      const Steinberg::Vst::MediaTypes mediaType, 
-      const Steinberg::Vst::BusDirection busDirection,
-      const Steinberg::Vst::BusType busType)
-   {
-      using namespace Steinberg;
-
-      unsigned channelsCount{0};
-
-      const auto busCount = component->getBusCount(mediaType, busDirection);
-      for(auto i = 0; i < busCount; ++i)
-      {
-         Vst::BusInfo busInfo;
-         if(component->getBusInfo(mediaType, busDirection, i, busInfo) == kResultOk)
-         {
-            if(busInfo.busType == busType)
-               channelsCount += busInfo.channelCount;
-         }
-      }
-      return channelsCount;
-   }
-}
-
-unsigned VST3Effect::GetAudioInCount() const
-{
-   return CountChannels(
-      mWrapper->mEffectComponent,
-      Steinberg::Vst::kAudio,
-      Steinberg::Vst::kInput,
-      Steinberg::Vst::kMain);
-}
-
-unsigned VST3Effect::GetAudioOutCount() const
-{
-   return CountChannels(
-      mWrapper->mEffectComponent,
-      Steinberg::Vst::kAudio,
-      Steinberg::Vst::kOutput,
-      Steinberg::Vst::kMain);
-}
-size_t VST3Effect::SetBlockSize(size_t maxBlockSize)
-{
-   mProcessingBlockSize = 
-      static_cast<Steinberg::int32>(std::min(maxBlockSize, mUserBlockSize));
-   return mProcessingBlockSize;
-}
-
-size_t VST3Effect::GetBlockSize() const
-{
-   return mProcessingBlockSize;
-}
-
-sampleCount VST3Effect::GetLatency() const
-{
-   if(mUseLatency)
-   {
-      if(!mRealtimeGroupProcessors.empty())
-         return mRealtimeGroupProcessors[0]->GetLatencySamples();
-      return mInitialDelay;
-   }
-   return { 0u };
-}
-
-bool VST3Effect::ProcessInitialize(
-   EffectSettings &settings, double sampleRate, ChannelNames)
-{
-   if(mWrapper->Initialize(settings, sampleRate, Steinberg::Vst::kOffline, mProcessingBlockSize))
-   {
-      mWrapper->ConsumeChanges(settings);
-      mInitialDelay = mWrapper->GetLatencySamples();
-      return true;
-   }
-   return false;
-}
-
-bool VST3Effect::ProcessFinalize() noexcept
-{
-   return GuardedCall<bool>([&]{
-      mWrapper->Finalize();
-      return true;
-   });
-}
-
-size_t VST3Effect::ProcessBlock(EffectSettings& settings, const float* const* inBlock, float* const* outBlock, size_t blockLen)
-{
-   return mWrapper->Process(inBlock, outBlock, blockLen);
-}
-
-bool VST3Effect::RealtimeInitialize(EffectSettings &settings, double sampleRate)
-{
-   return true;
-}
-
-bool VST3Effect::RealtimeAddProcessor(
-   EffectSettings &settings, unsigned numChannels, float sampleRate)
-{
-   using namespace Steinberg;
-
-   try
-   {
-      auto wrapper = std::make_unique<VST3Wrapper>(*mModule, mEffectClassInfo.ID());
-      if(!wrapper->Initialize(settings, sampleRate, Vst::kRealtime, mProcessingBlockSize))
-         throw std::runtime_error { "VST3 realtime initialization failed" };
-      wrapper->ConsumeChanges(settings);
```

Comment:
This logic moves to `RealtimeInitialize` while the new `RealtimeAddProcessor` does nothing.  The `ConsumeChanges` is lost so I am again unsure that the processor really begins in the correct state.